### PR TITLE
sd-mux-ctrl: add shell completions

### PR DESCRIPTION
**Pull request by newAM**
## Description of changes

I also added myself as a maintainer in a separate commit.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sarcasticadmin** [comment]
Result of `nixpkgs-review pr 333120` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sd-mux-ctrl</li>
  </ul>
</details>

**Review by sarcasticadmin — APPROVED** [review]

**Comment by newAM** [comment]
Thanks for the review @sarcasticadmin!
If you think this PR is ready package maintainers can now merge with `@NixOS/nixpkgs-merge-bot merge`

**Comment by sarcasticadmin** [comment]
@NixOS/nixpkgs-merge-bot merge

**Comment by sarcasticadmin** [comment]
> Thanks for the review @sarcasticadmin! If you think this PR is ready package maintainers can now merge with `@NixOS/nixpkgs-merge-bot merge`

Well that's great to know, thank you! I wasn't aware that was available :slightly_smiling_face: Looks like its not enabled for pkgs outside of `pkgs/by-name`, but I'm sure someone will see this soon.

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/tools/misc/sd-mux-ctrl/default.nix [hunk 1]
```diff
@@ -10,19 +10,21 @@ stdenv.mkDerivation rec {
...
 
   postInstall = ''
     install -D -m 644 ../doc/man/sd-mux-ctrl.1 $out/share/man/man1/sd-mux-ctrl.1
+    installShellCompletion --cmd sd-mux-ctrl \
+      --bash ../etc/bash_completion.d/sd-mux-ctrl
   '';
 
   meta = with lib; {
     description = "Tool for controlling multiple sd-mux devices";
     homepage = "https://wiki.tizen.org/SD_MUX";
```
Comment (Aleksanaa) [inline pkgs/tools/misc/sd-mux-ctrl/default.nix:25]: Unable to access this link now

Comment (sarcasticadmin) [reply pkgs/tools/misc/sd-mux-ctrl/default.nix:25]: It loads for me, just very slowly.

**Comment by newAM** [comment]
> Looks like its not enabled for pkgs outside of `pkgs/by-name`, but I'm sure someone will see this soon.

New tool, I didn't know that either!  I opened #333140 to move this package over there for next time.

File: pkgs/tools/misc/sd-mux-ctrl/default.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/tools/misc/sd-mux-ctrl/default.nix:25]: I can't even resolve that domain, not sure

```
=> dig wiki.tizen.org

; <<>> DiG 9.18.27 <<>> wiki.tizen.org
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 47343
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;wiki.tizen.org.                        IN      A

;; Query time: 2 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Thu Aug 08 10:50:28 CST 2024
;; MSG SIZE  rcvd: 43
```

**Pull request merged**